src/utils/types/object/object.cpp

```diff
@@ -0,0 +1,106 @@
+#include "object.h"
+
+Object::Object() {
+    setInvalid();
+}
+
+Object::~Object() {
+
+}
+
+Position Object::getPosition() {
+    Position retn = _position;
+
+    return retn;
+}
+
+Velocity Object::getVelocity() {
+    Velocity retn = _velocity;
+
+    return retn;
+}
+
+Angle Object::getOrientation() {
+    Angle retn = _orientation;
+
+    return retn;
+}
+
+bool Object::isObjectLoss() {
+    return _lossFilter.checkLoss();
+}
+
+bool Object::isObjectSafe() {
+    return _noiseFilter.checkNoise();
+}
+
+void Object::updateObject(float confidence, Position pos, Angle orientation) {
+    // Update confidence
+    _confidence = confidence;
+
+    // If pos is invalid (robot is not visible in frame)
+    if(pos.isInvalid()) {
```

Comment:
This is the decision tree that runs all the three filters:

    If pos is invalid (robot is not visible in frame):
        Start loss if not started
            If is loss, set object as invalid and reset noise
            If is not loss yet, calls kalman to update object position (using predict)
    Else if position is valid
        Start noise if not initialized and set invalid
            If is object safe, reset loss and update kalman matrixes with position
            If is not safe, maintain object as invalid (not safe) and reset loss.